### ISSUE TITLE
Extract Block Models from Unity Directly

### ISSUE THREAD

**Issue by PeteMichaud** [ApanLoon/EmpyrionStuff#45]
Right now block models are created from scratch in code. The nice thing is that it makes the program totally independent. The bad things are numerous:

1. It's error prone, which is proven by the open issue about normals being bad
2. It's a maintenance nightmare
3. It requires someone to manually create all possible block models, which is not realistic and will not stay up to date
4. The block that are special, like the turrets for example, are not displayed as they would be in the game, but rather as generic full blocks that are sometimes the wrong size

My suggestion is to extract the models from the game itself. Extracting unity assets is totally possible, and I bet we could more or less automate the process of extracting the models and mapping them to their respective BlockType codes, which would only need to be done once per new release version of the game. 

One option is to do this at build time and include the assets in the application download. The other option is to do this at runtime and require that the end user has Empyrion installed on their machine. 

I think doing it at build time is the best option. One thing I'd want to do before attempting this is to get Eleon's blessing for the extraction. They might even be willing to help if they see the project as a potential value add for the modding community and therefore popularity booster for the game.

**Comment by ApanLoon** [comment]
Yes, I agree. The current method is not great. It is not all bad though since I have a script that generates that huge source code file out of an OBJ for me.

When I started this project I never intended to make visually accurate blocks - just cubes was the limit of my intentions. The reason for this is mostly based on the copyright issue. The only way I see this happening is to read the assets at run-time and this may make the program slow to load. 

Yesterday Taelyn was really helpful giving me a list of all the shape names for each block type, I am not sure if he knows what I wanted the list for, but I am hoping that he does and that this implies that Eleon is ok with the project in its current form, but using their assets directly is most likely a different story altogether. I suppose that one could ask, but for now, I think it is best to stick to my ugly models. :-)

**Comment by ApanLoon** [comment]
Would it be a useful step to make the code read the ASCII OBJ file instead of relying on generating a huge .cs file to compile?
